### ISSUE TITLE
Git: fatal: Could not read from remote repository.

### ISSUE THREAD

**Issue by space55** [concourse/concourse#5823]
## Summary

When cloning a repo through the Git resource (that the SSH key has access to, I checked by cloning using the same SSH key), I receive:

```
run check step: run check step: check: resource script '/opt/resource/check []' failed: exit status 128

stderr:
Identity added: /tmp/git-resource-private-key (ubuntu@ip-172-31-36-103)
Cloning into '/tmp/git-resource-repo-cache'...
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

## Steps to reproduce

1. Run a fresh Concourse instance, with docker-compose
2. Create a sample pipeline, f.ex, one that clones the Concourse repo, then echo's "Hello" to the console
3. The concourse instance will fail

## Expected results

Should not receive that error

## Actual results

Fails to clone pipeline

## Additional context

Sample pipeline:

```yaml
resources:
  - name: repo
    type: git
    source:
      uri: git@github.com:concourse/concourse.git
      branch: master
      private_key: ((git-key))

  - name: Run tests
    plan:
      - get: repo
        trigger: true
      - task: run the test suite
        file: repo/ci/tasks/run-tests.yml
```

## Triaging info

* Concourse version: Fresh 6.3.0 install on AWS EC2
* Browser (if applicable): Chrome, but I don't think that applies
* Did this used to work? Never worked on this fresh install, works on my other clusters, some running the same version

Minor additional discussion going on in https://discuss.concourse-ci.org/t/resource-script-opt-resource-check-failed-exit-status-128/2310

**Comment by jamieklassen** [comment]
**tl;dr** can you provide the `docker-compose.yml` you're using?

When I use the precise YAML you provided, I get this:
```bash
$ fly -t dev sp -p pipeline -c pipeline.yml -n
resources:
  resource repo has been added:
+ name: repo
+ source:
+   branch: master
+   private_key: ((git-key))
+   uri: git@github.com:concourse/concourse.git
+ type: git

  resource Run tests has been added:
+ name: Run tests
+ source: null
+ type: ""

error: invalid pipeline config:
malformed config: error unmarshaling JSON: while decoding JSON: json: unknown field "plan"
```

which is probably just due to a missing `jobs:` section. Adding that, I'm able to set the pipeline, and very soon after I unpause it, I get this:

```bash
$ fly -t dev resources -p pipeline --json
[
  {
    "name": "repo",
    "pipeline_name": "pipeline",
    "team_name": "main",
    "type": "git",
    "failing_to_check": true,
    "check_setup_error": "undefined vars: git-key"
  }
]
```

so the key piece of information I'm missing here is - where is concourse getting that `((git_key))` from? Is it just `fly set-pipeline` with `-v git_key="<a literal SSH private key>"`? Or are you using a credential manager?

In the linked discussion you mention using AWS Secrets Manager -- maybe that's relevant? Overall, I'd just like to know a bit more about how your underlying Concourse is configured, because my instinct was to interpret "fresh" as "vanilla", i.e. just using exactly http://concourse-ci.org/docker-compose.yml. But I think you have a nontrivial configuration and by "fresh" you meant "with a brand-new VM".
